src/components/interface/filter/index.jsx

```diff
@@ -1,69 +1,137 @@
 import React from 'react';
-import Button from '@material-ui/core/Button';
-import FilterIcon from '@material-ui/icons/PlaylistAdd';
-import EnhancedMenu from './EnhancedMenu';
+import PropTypes from 'prop-types';
+
+import { withStyles } from '@material-ui/core/styles';
+import Input from '@material-ui/core/Input';
+import LensIcon from '@material-ui/icons/Lens';
+import SearchIcon from '@material-ui/icons/Search';
+
+import CheckBox from '../CheckBox';
+import db from '../../../firebase/firebase';
+import FilterMenu from './FilterMenu';
+import styles from '../../styles';
 
 class Filter extends React.Component {
-  state = {
-    anchorEl: null
-  };
+  constructor(props) {
+    super(props);
+    this.state = {
+      options: {},
+      filtered: {
+        academicLevel: [],
+        awardConsideration: [],
+        competitionLevel: [],
+        danceSize: [],
+        danceStyle: [],
+        school: []
+      }
+    };
+  }
 
-  handleClick = (event) => {
-    this.setState({ anchorEl: event.currentTarget });
-  };
+  componentDidMount() {
+    this.subscribe = db.collection('settings').onSnapshot((querySnapshot) => {
+      const options = {};
+      querySnapshot.forEach((doc) => {
+        options[doc.id] = Object.keys(doc.data());
+      });
+      this.setState({ options });
+    });
+  }
 
-  handleClose = () => {
-    this.setState({ anchorEl: null });
-  };
+  componentDidUpdate(prevProps, prevState) {
+    const { filtered: prevFiltered } = prevState;
+    const { filtered: currFiltered } = this.state;
+    const { handleFilters } = this.props;
+    if (JSON.stringify(prevFiltered) !== JSON.stringify(currFiltered)) {
```

Comment:
Is there a way we could do this without JSON.stringify. Does Lodash have an object comparator? This feels a bit hacky.